### ISSUE TITLE
Atomtyping benchmarks

### ISSUE THREAD

**Issue by mattwthompson** [mosdef-hub/foyer#132]
To better understand how atomtyping can be sped up, we should create some benchmarks. Some variables that would be worth exploring

* System size: scaling from a few hundred or thousand to hundreds of thousands or millions of atoms

* Force field size, i.e. an xml file that only contains tip3p water vs. the entire OPLS force field

* Sparse vs. dense bond networks, i.e. water vs a slab of hexa-coordinated atoms in a crystal

Might be better to store this code elsewhere & I can contribute some systems

**Comment by chrisiacovella** [comment]
Good idea.  As Andrew and I just discovered, forcefield size seems to play a significant role as the system size increases. 

 As we've discussed before, we should have a "reduce" function that outputs just the relevant parameters/rules used for a given system (into an xml file).  This would be useful for:

- Publishing papers, as it allows just the parameters used in a paper to be provided

- Speeding up large systems, as you could run the "reduce" function on a small system, then use that forcefield file for the larger system.
